app/models/concerns/audit_search.rb

```diff
@@ -0,0 +1,73 @@
+
+    scoped_search :relation => :search_parameters, :on => :name, :complete_value => true, :rename => :parameter, :only_explicit => true
+    scoped_search :relation => :search_templates, :on => :name, :complete_value => true, :rename => :provisioning_template, :only_explicit => true
+    scoped_search :relation => :search_ptables, :on => :name, :complete_value => true, :rename => :partition_table, :only_explicit => true
+    scoped_search :relation => :search_os, :on => :name, :complete_value => true, :rename => :os, :only_explicit => true
+    scoped_search :relation => :search_os, :on => :title, :complete_value => true, :rename => :os_title, :only_explicit => true
+    scoped_search :relation => :search_class, :on => :name, :complete_value => true, :rename => :puppetclass, :only_explicit => true
+    scoped_search :relation => :search_hosts, :on => :name, :complete_value => true, :rename => :host, :only_explicit => true
+    scoped_search :relation => :search_hostgroups, :on => :name, :complete_value => true, :rename => :hostgroup, :only_explicit => true
+    scoped_search :relation => :search_hostgroups, :on => :title, :complete_value => true, :rename => :hostgroup_title, :only_explicit => true
+    scoped_search :relation => :search_users, :on => :login, :complete_value => true, :rename => :user, :only_explicit => true
+    scoped_search :relation => :search_nics, :on => :name, :complete_value => true, :rename => :interface_fqdn, :only_explicit => true
+    scoped_search :relation => :search_nics, :on => :ip, :complete_value => true, :rename => :interface_ip, :only_explicit => true
+    scoped_search :relation => :search_nics, :on => :mac, :complete_value => true, :rename => :interface_mac, :only_explicit => true
+  end
+
+  module ClassMethods
+    def auditable_type_complete_values
+      # Note: this will only work properly in production, as models are lazy loaded
+      # in development, meaning we won't have all the class names when this runs.
+      complete_values = audited_classes_without_sti.each_with_object({}) do |name, obj|
+        obj[name.underscore.to_sym] = name
+      end
+      # Add aliases and STI handling
+      complete_values.merge!(
+        :compute_resource => 'ComputeResource',
+        :host => 'Host::Base',
+        :interface => 'Nic::Base',
+        :location => 'Location',
+        :os => 'Operatingsystem',
+        :organization => 'Organization',
+        :override_value => 'LookupValue',
+        :partition_table => 'Ptable',
+        :smart_class_parameter => 'PuppetclassLookupKey',
+        :smart_variable => 'VariableLookupKey',
+        :parameter => 'Parameter'
+      )
+    end
+
+    private
+
+    #This is a workaround needed until we get proper STI auditing
+    def audited_classes_without_sti
+      audited_class_names.reject do |name|
+        klass = name.constantize
+        name.include?(':') ||
```

Comment:
wouldn't this reject all namespaced classes, e.g. `ForemanOpenscap::ArfReport` or `Katello::HostCollection`? Also if this should remain, `::` would be more descriptive I think.

Reply:
Good point, i'll fix this.